README
NAME
    Dist::Zilla::Plugin::ApocalypseTests - Creates the Test::Apocalypse
    testfile for Dist::Zilla

VERSION
      This document describes v1.001 of Dist::Zilla::Plugin::ApocalypseTests - released March 05, 2011 as part of Dist-Zilla-Plugin-ApocalypseTests.

DESCRIPTION
    This is an extension of Dist::Zilla::Plugin::InlineFiles, providing the
    following files:

    *   t/apocalypse.t - Runs the dist through Test::Apocalypse

        For more information on what the test does, please look at
        Test::Apocalypse.

                # In your dist.ini:
                [ApocalypseTests]

ATTRIBUTES
  allow
    This option will be passed directly to Test::Apocalypse to control which
    sub-tests you want to run.

    The default is nothing.

  deny
    This option will be passed directly to Test::Apocalypse to control which
    sub-tests you want to run.

    The default is nothing.

SEE ALSO
    Please see those modules/websites for more information related to this
    module.

    *   Dist::Zilla

    *   Test::Apocalypse

SUPPORT
  Perldoc
    You can find documentation for this module with the perldoc command.

      perldoc Dist::Zilla::Plugin::ApocalypseTests

  Websites
    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

    *   Search CPAN

        <http://search.cpan.org/dist/Dist-Zilla-Plugin-ApocalypseTests>

    *   RT: CPAN's Bug Tracker

        <http://rt.cpan.org/NoAuth/Bugs.html?Dist=Dist-Zilla-Plugin-Apocalyp
        seTests>

    *   AnnoCPAN: Annotated CPAN documentation

        <http://annocpan.org/dist/Dist-Zilla-Plugin-ApocalypseTests>

    *   CPAN Ratings

        <http://cpanratings.perl.org/d/Dist-Zilla-Plugin-ApocalypseTests>

    *   CPAN Forum

        <http://cpanforum.com/dist/Dist-Zilla-Plugin-ApocalypseTests>

    *   CPANTS Kwalitee

        <http://cpants.perl.org/dist/overview/Dist-Zilla-Plugin-ApocalypseTe
        sts>

    *   CPAN Testers Results

        <http://cpantesters.org/distro/D/Dist-Zilla-Plugin-ApocalypseTests.h
        tml>

    *   CPAN Testers Matrix

        <http://matrix.cpantesters.org/?dist=Dist-Zilla-Plugin-ApocalypseTes
        ts>

  Email
    You can email the author of this module at "APOCAL at cpan.org" asking
    for help with any problems you have.

  Internet Relay Chat
    You can get live help by using IRC ( Internet Relay Chat ). If you don't
    know what IRC is, please read this excellent guide:
    <http://en.wikipedia.org/wiki/Internet_Relay_Chat>. Please be courteous
    and patient when talking to us, as we might be busy or sleeping! You can
    join those networks/channels and get help:

    *   irc.perl.org

        You can connect to the server at 'irc.perl.org' and join this
        channel: #perl-help then talk to this person for help: Apocalypse.

    *   irc.freenode.net

        You can connect to the server at 'irc.freenode.net' and join this
        channel: #perl then talk to this person for help: Apocal.

    *   irc.efnet.org

        You can connect to the server at 'irc.efnet.org' and join this
        channel: #perl then talk to this person for help: Ap0cal.

  Bugs / Feature Requests
    Please report any bugs or feature requests by email to
    "bug-dist-zilla-plugin-apocalypsetests at rt.cpan.org", or through the
    web interface at
    <http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Dist-Zilla-Plugin-Apocal
    ypseTests>. You will be automatically notified of any progress on the
    request by the system.

  Source Code
    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    <http://github.com/apocalypse/perl-dist-zilla-plugin-apocalypsetests>

      git clone git://github.com/apocalypse/perl-dist-zilla-plugin-apocalypsetests.git

AUTHOR
    Apocalypse <APOCAL@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2011 by Apocalypse.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

    The full text of the license can be found in the LICENSE file included
    with this distribution.



Changes
Revision history for Dist::Zilla::Plugin::ApocalypseTests

1.001
	Released: 2011-03-06 00:37:26 UTC

	Bumped the Test::Apocalypse prereq to 1.000

1.000
	Released: 2010-12-14 03:40:32 UTC

	Converted to Dist::Zilla for the release process!

0.01

	initial release


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  'build_requires' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'Module::Build' => '0.3601',
    'Test::More' => '0.88'
  },
  'configure_requires' => {
    'Module::Build' => '0.3601'
  },
  'dist_abstract' => 'Creates the Test::Apocalypse testfile for Dist::Zilla',
  'dist_author' => [
    'Apocalypse <APOCAL@cpan.org>'
  ],
  'dist_name' => 'Dist-Zilla-Plugin-ApocalypseTests',
  'dist_version' => '1.001',
  'license' => 'perl',
  'module_name' => 'Dist::Zilla::Plugin::ApocalypseTests',
  'recommends' => {},
  'recursive_test_files' => 1,
  'requires' => {
    'Dist::Zilla::Plugin::InlineFiles' => '2.101170',
    'Dist::Zilla::Role::FileMunger' => '2.101170',
    'Moose' => '1.03',
    'perl' => '5.006'
  },
  'script_files' => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

use strict;
use warnings;

BEGIN { require 5.006; }

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'Creates the Test::Apocalypse testfile for Dist::Zilla',
  'AUTHOR' => 'Apocalypse <APOCAL@cpan.org>',
  'BUILD_REQUIRES' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'ExtUtils::MakeMaker' => '6.31',
    'Test::More' => '0.88'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31'
  },
  'DISTNAME' => 'Dist-Zilla-Plugin-ApocalypseTests',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Dist::Zilla::Plugin::ApocalypseTests',
  'PREREQ_PM' => {
    'Dist::Zilla::Plugin::InlineFiles' => '2.101170',
    'Dist::Zilla::Role::FileMunger' => '2.101170',
    'Moose' => '1.03'
  },
  'VERSION' => '1.001',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



